File: 01-programming/csharp/concepts/code/01-basics/conditions-if-else.cs
// -o-o-o-o- Conditions (If else) -o-o-o-o-

/* This file goes over how to do a basic if else statement in C# simply.
 *
 *  # # # # KEY # # # #
 *
 * \n - creates a new line (also Console.WriteLine() does this automatically)
 *
 * != - not equal to operator
 * == - equal to operator
 *
 * Note: = is the assign operator, it is used to assign values to variables.
 * == means equal to and is used for conditional statements. Never use a single =
 * to check for conditions such as if a variable is equal to a certain value.
 *
 * Console.WriteLine() - prints text to the console with a newline at the end.
 *                       equivalent to printf("...\n") in C.
 *
 * Console.Write()     - prints text WITHOUT a newline at the end.
 *                       equivalent to printf("...") in C (no \n).
 *
 * Console.ReadLine()  - reads a line of text input from the user as a string.
 *                       equivalent to scanf() in C.
 *
 * int.Parse()         - converts a string into an integer.
 *                       needed because Console.ReadLine() always returns a string.
 *                       eg. int a = int.Parse(Console.ReadLine());
 *
 * int                 - declares an integer variable, same as in C.
 *
 * */

using System;

class Program {
    static void Main() {

        int a;
        Console.Write("Please enter either 0 or 1: ");
        a = int.Parse(Console.ReadLine()); // Gets user input and converts it to an int

        // reads as -> If a is not equal to 0 -> run code in {}
        if (a != 0) {
            Console.WriteLine("Variable A is equal to 1. You get a treat");
        }

        // checks if the above if statement is true and only runs if it is not true.
        else {
            Console.WriteLine("Variable A is not equal to 1. You get no treats");
        }

    }
}

File: 01-programming/python/concepts/code/01-basics/conditions-if-else.py
# -o-o-o-o- Conditions (If Else) -o-o-o-o-

# This file goes over how to do basic if else statements in Python simply.
#
#  # # # # KEY # # # #
#
# if       - checks a condition and runs the indented block if it is true.
#            No parentheses required around the condition (unlike C and C#).
#            A colon : is required at the end of the if line.
#            C:  if (a != 0) { ... }
#            C#: if (a != 0) { ... }
#            Py: if a != 0:
#
# elif     - short for "else if". Checks another condition if the previous
#            was false. Equivalent to else if in C and C#.
#
# else     - runs if none of the above conditions were true.
#            Same concept as C and C#. Ends with a colon.
#
# Indentation - Python uses indentation (4 spaces) to define code blocks.
#            There are NO curly braces { }. Getting indentation wrong causes
#            an IndentationError. This replaces { } from C and C#.
#
# !=       - not equal to operator. Same as C and C#.
# ==       - equal to operator. Same as C and C#.
# >  <     - greater than / less than. Same as C and C#.
# >= <=    - greater/less than or equal to. Same as C and C#.
# and      - logical AND. Equivalent to && in C and C#.
# or       - logical OR.  Equivalent to || in C and C#.
# not      - logical NOT. Equivalent to !  in C and C#.
#
# input()  - reads a line of user input. Always returns a string.
#            Equivalent to scanf() in C and Console.ReadLine() in C#.
# int()    - converts a string to an integer.
#            eg. int(input("Enter: "))
#
# print()  - prints text to the console with a newline at the end.
#

a = int(input("Please enter either 0 or 1: "))

# reads as -> if a is not equal to 0 -> run the indented block
if a != 0:
    print("Variable a is equal to 1. You get a treat")

# checks if the above condition was false and only runs if it was not true
else:
    print("Variable a is not equal to 1. You get no treats")


# --- elif example ---

score = int(input("\nEnter a score (0-100): "))

if score >= 90:
    print("Grade: A")
elif score >= 80:
    print("Grade: B")
elif score >= 70:
    print("Grade: C")
elif score >= 60:
    print("Grade: D")
else:
    print("Grade: F")


# --- Combining conditions ---

x = int(input("\nEnter a number: "))

if x > 0 and x % 2 == 0:
    print("Positive and even")
elif x > 0 and x % 2 != 0:
    print("Positive and odd")
elif x < 0:
    print("Negative number")
else:
    print("Zero")
